src/workspace.ts

```diff
@@ -63,70 +91,98 @@ export interface WorkspaceRepo {
   github: GitHubRepo;
 }
 
-export interface WorkspaceInitOptions {
+export interface WorkspaceOptions {
+  token: string;
+  dir: string;
+  match: string[];
+  exclude?: string[];
   fresh?: boolean;
   verbose?: boolean;
 }
 
+/**
+ * Workspace - An instance for creating new workspaces. The `init()`
+ * method drives the loading, creation, and configuration of each workspace
+ * repo. Repos are loaded from GitHub, and a GitHub API Token is required to
+ * use.
+ *
+ * A Workspace instance returns WorkspaceRepo objects, which the user
+ * can use to interact with each repo in the workspace.
+ */
 export class Workspace {
-  dir: string;
-  private _github: GitHubConnection;
-  private _initializedRepos: WorkspaceRepo[]|undefined;
+  readonly dir: string;
+  private readonly options: WorkspaceOptions;
+  private readonly _github: GitHubConnection;
 
-  constructor(options: {token: string, dir: string}) {
+  constructor(options: WorkspaceOptions) {
     this.dir = options.dir;
+    this.options = options;
     this._github = new GitHubConnection(options.token);
   }
 
   /**
-   * Helper function that returns true when workspace.init() has been run.
+   * Initialize the workspace. This is the driver of all initialization and
+   * setup logic.
    */
-  get isInitialized() {
-    return !!this._initializedRepos;
+  async init(): Promise<
```

Comment:
maybe rename to `getRepositories`?

Reply:
I think that loses the "only call once" implication. Lets get this PR merged and discuss in future PR (for this breaking change or the next one)